src/parsers/manifestjson.js

```diff
@@ -281,13 +300,13 @@ export default class ManifestJSONParser extends JSONParser {
     // Arrays can be extremely verbose, this tries to make them a little
     // more sane. Using a regex because there will likely be more as we
     // expand the schema.
-    const match = error.dataPath.match(
-      /^\/(permissions|optional_permissions)\/([\d+])/
-    );
+    const match = error.dataPath.match(PERMS_DATAPATH_REGEX);
+
     if (
       match &&
       baseObject.code !== messages.MANIFEST_BAD_PERMISSION.code &&
-      baseObject.code !== messages.MANIFEST_BAD_OPTIONAL_PERMISSION.code
+      baseObject.code !== messages.MANIFEST_BAD_OPTIONAL_PERMISSION.code &&
+      baseObject.code !== messages.MANIFEST_PERM_UNSUPPORTED
```

Comment:
shouldn't that be `messages.MANIFEST_PERM_UNSUPPORTED.code`?

Reply:
As for some other messages that are being created using an exported method, `MANIFEST_PERM_UNSUPPORTED` is a string and not an object, and there is a function paired with that rule code `const` named `manifestPermUnsupported` which is the one returning the object with the code property.

There are more that follows this slightly different convention, e.g.:
- `MANIFEST_CSP` and `manifestCsp`
- `MANIFEST_CSP_UNSAFE_EVAL` and `manifestCspUnsafeEval`
- `MANIFEST_ICON_NOT_FOUND` and `manifestIconMissing`

To be honest, I also find the mix of these two slightly different conventions quite confusing, but I think we should choose a convention that we like and use it for all of the rules not just newly ones added, I don't see much value in that.

Given how big this PR was already becoming, I opted to just follow the current conventions as is for the new ones added or changed in this PR.

Reply:
ok